calour/ms1_experiment.py

```diff
@@ -0,0 +1,168 @@
+        if feature_field == 'gnps':
+            if 'gnps' not in self.feature_metadata.columns:
+                logger.warn('no gnps feature metadata present. Plase run ".prepare_gnps()" to see gnps labels.')
+                feature_field = 'id'
+        # plot the experiment using taxonmy field and dbbact database
+        super().plot(feature_field=feature_field, databases=databases, **kwargs)
+
+    @Experiment._record_sig
+    def sort_mz(exp, inplace=False):
+        '''Sort the features based on the taxonomy
+
+        Sort features based on the taxonomy (alphabetical)
+
+        Parameters
+        ----------
+        inplace : bool (optional)
+            False (default) to create a copy
+            True to Replace data in exp
+        Returns
+        -------
+        exp : Experiment
+            sorted by taxonomy
+        '''
+        logger.debug('sorting by taxonomies')
+        taxonomy = _get_taxonomy_string(exp, remove_underscore=True)
+        sort_pos = np.argsort(taxonomy, kind='mergesort')
+        exp = exp.reorder(sort_pos, axis=1, inplace=inplace)
+        return exp
+
+    def plot_sort(self, fields=None, feature_field='gnps', sample_color_bars=None, feature_color_bars=None,
+                  gui='cli', databases=('gnps',), color_bar_label=True, **kwargs):
+        '''Plot bacteria after sorting by field
+
+        This is a convenience wrapper for plot()
+
+        Parameters
+        ----------
+        fields : str or list of str or None (optional)
+            The field to sort samples by before plotting
+            If list of str, sort by each field according to order in list
+            if None, do not sort
+        sample_color_bars : list, optional
+            list of column names in the sample metadata. It plots a color bar
+            for each column. It doesn't plot color bars by default (``None``)
+        feature_color_bars : list, optional
+            list of column names in the feature metadata. It plots a color bar
+            for each column. It doesn't plot color bars by default (``None``)
+        color_bar_label : bool, optional
+            whether to show the label for the color bars
+        gui : str or None, optional
+            GUI to use:
+            'cli' : simple command line gui
+            'jupyter' : jupyter notebook interactive gui
+            'qt5' : qt5 based interactive gui
+            None : no interactivity - just a matplotlib figure
+        databases : Iterable of str
+            a list of databases to access or add annotation
+        kwargs : dict, optional
+            keyword arguments passing to :ref:`plot<plot-ref>` function.
+
+        '''
+        if feature_field == 'gnps':
+            if 'gnps' not in self.feature_metadata.columns:
+                logger.warn('no gnps feature metadata present. Plase run ".prepare_gnps()" to see gnps labels.')
+                feature_field = 'id'
+        if fields is not None:
```

Comment:
did super cool solution to all the overrides.
instead, each experiment subclass has parameters for the default databases and feature_field.
And we use this from the main plot functions if the value of databases/feature_field is False
so no more overrides!